tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/ScopedBuilder.scala

```diff
@@ -141,4 +149,68 @@ class ScopedBuilder
         )
     )
   }
+
+  /**
+   * [f EXCEPT ![a1] = e1, ![a2] = e2 ... ![an] = en]
+   *
+   * Is equivalent to {{{[[f EXCEPT ![a1] = e1] EXCEPT ![a2] = e2] EXCEPT ... ![an] = en]}}}
+   */
+  def multiPointExcept(
+      f: TBuilderInstruction,
+      args: (TBuilderInstruction, TBuilderInstruction)*): TBuilderInstruction = {
+    require(args.nonEmpty)
+    args.foldLeft(f) { case (fn, (ai, ei)) =>
+      except(fn, ai, ei)
+    }
+  }
+
+  /**
+   * [f EXCEPT ![a1][a2][...][an] = e]
+   *
+   * Is equivalent to {{{[f EXCEPT ![a1] = [f[a1] EXCEPT ![a2] = [ ... EXCEPT ![an] = e]]]}}}
+   */
+  def multiDepthExcept(
+      f: TBuilderInstruction,
+      e: TBuilderInstruction,
+      args: TBuilderInstruction*): TBuilderInstruction = {
+    require(args.nonEmpty)
+
+    args match {
+      case Seq(head) => except(f, head, e)
+      case head +: tail =>
+        except(
+            f,
+            head,
+            multiDepthExcept(app(f, head), e, tail: _*),
+        )
+    }
+  }
+
+  /**
+   * [f EXCEPT ![a1][a2][...][an] = ea, ![b1][b2][...][bn] = eb, ..., ![z1][z2][...][zn] = ez]
+   *
+   * @param args
+   *   Pairs of the shape (ei, Seq(i1, ..., in))
+   */
+  def genericExcept(
```

Comment:
```suggestion
  def exceptGeneral(
```

Reply:
Shall we also add a convenience method for single-point shallow except, which is by far the most common case? Or if `genericExcept` could be renamed to `except`?

Reply:
Isn't single-point shallow except just `except`?